creativecloud/blocks/sidenav/sidenav.js

```diff
@@ -7,24 +7,39 @@ import '../../deps/merch-sidenav.js';
 const CATEGORY_TYPE_PREFIX = 'categories/';
 const TYPE_TYPE_PREFIX = 'types/';
 
-const getCategories = (arrayCategories) => {
-  const tag = createTag('sp-sidenav', { variant: 'multilevel', manageTabIndex: true });
+const getValue = (id) => (id?.substring(id.lastIndexOf('/') + 1) || id);
+
+const getCategories = (items, isMultilevel, mapCategories) => {
+  const configuration = { manageTabIndex: true };
+  if (isMultilevel) {
+    configuration.variant = 'multilevel';
+  }
+  const mapParents = [];
+  const tag = createTag('sp-sidenav', configuration);
   const merchTag = createTag('merch-sidenav-list', { deeplink: 'category' });
   merchTag.append(tag);
-  const mapParents = {};
-  arrayCategories.forEach((item) => {
-    if (item.name?.length > 0) {
-      const id = item.id.slice(CATEGORY_TYPE_PREFIX.length);
-      const value = id.substring(id.lastIndexOf('/') + 1) || id;
-      const isParent = id.split('/').length === 1;
+  items.forEach((item) => {
+    if (item) {
+      let parent = tag;
+      const value = getValue(item.id);
+      const isParent = item.id.split('/').length <= 2;
```

Comment:
Shouldn't this be checking for an array length of `1`? Otherwise something like `categories/photography` would be considered a parent.

Reply:
"thing" is there is always "all" that is there, may be i should i add a comment?